framework/auth/views.py

```diff
@@ -556,17 +553,25 @@ def unconfirmed_email_add(auth=None):
     }, 200
 
 
-def send_confirm_email(user, email, external_id_provider=None, external_id=None):
+def send_confirm_email(user, email, renew=False, external_id_provider=None, external_id=None):
```

Comment:
when user asks to resend confirmation email, refresh the token with `renew = True`